tests/Meziantou.Framework.CodeOwners.Tests/CodeOwnersParserTests.cs

```diff
@@ -180,4 +180,86 @@ public void ParseCodeOwnersWithPatternsWithoutMembers()
+    }
+
+    [Fact]
+    public void ParseCodeOwnersWithDefaultOwners()
+    {
+        var actual = CodeOwnersParser.Parse("[Test] @defaultOwner default.owner@example.com\n*").ToArray();
+
+        var expected = new CodeOwnersEntry[]
+        {
+            CodeOwnersEntry.FromUsername(0, "*", "defaultOwner", section: new CodeOwnersSection("Test", 1, ["@defaultOwner", "default.owner@example.com"])),
+            CodeOwnersEntry.FromEmailAddress(0, "*", "default.owner@example.com", section: new CodeOwnersSection("Test", 1, ["@defaultOwner", "default.owner@example.com"])),
+        };
+
+        actual.Should().Equal(expected);
+    }
+
+    [Fact]
+    public void ParseCodeOwnersWithDefaultOwnersOverriden()
+    {
+        var actual = CodeOwnersParser.Parse("[Test] @defaultOwner default.owner@example.com\n* @user1 @user2").ToArray();
+
+        var expected = new CodeOwnersEntry[]
+        {
+            CodeOwnersEntry.FromUsername(0, "*", "user1", section: new CodeOwnersSection("Test", 1, ["@defaultOwner", "default.owner@example.com"])),
+            CodeOwnersEntry.FromUsername(0, "*", "user2", section: new CodeOwnersSection("Test", 1, ["@defaultOwner", "default.owner@example.com"])),
+        };
+
+        actual.Should().Equal(expected);
+    }
+
+    [Fact]
+    public void ParseCodeOwnersWithRequiredReviewerCountAndDefaultOwners()
+    {
+        var actual = CodeOwnersParser.Parse("[Test][2] @defaultOwner default.owner@example.com\n*").ToArray();
+
+        var expected = new CodeOwnersEntry[]
+        {
+            CodeOwnersEntry.FromUsername(0, "*", "defaultOwner", section: new CodeOwnersSection("Test", 2, ["@defaultOwner", "default.owner@example.com"])),
+            CodeOwnersEntry.FromEmailAddress(0, "*", "default.owner@example.com", section: new CodeOwnersSection("Test", 2, ["@defaultOwner", "default.owner@example.com"])),
+        };
+
+        actual.Should().Equal(expected);
+    }
+
+    [Fact]
+    public void ParseSectionWithCRLFLineEndings()
+    {
+        const string Content = "\r\n" +
+                               "[Test1]\r\n" +
+                               "* @user1\r\n" +
+                               "\r\n" +
+                               "[Test2][2]\r\n" +
+                               "* @user2\r\n" +
+                               "\r\n" +
+                               "[Test3] @defaultOwner\r\n" +
+                               "* @user3\r\n" +
+                               " ";
+
+        var actual = CodeOwnersParser.Parse(Content).ToArray();
+
+        var expected = new CodeOwnersEntry[]
+        {
+            CodeOwnersEntry.FromUsername(0, "*", "user1", section: new CodeOwnersSection("Test1")),
+            CodeOwnersEntry.FromUsername(1, "*", "user2", section: new CodeOwnersSection("Test2", 2)),
+            CodeOwnersEntry.FromUsername(2, "*", "user3", section: new CodeOwnersSection("Test3", 1, ["@defaultOwner"])),
+        };
+
+        actual.Should().Equal(expected);
+    }
```

Comment:
What's the behavior of GitLab in the following cases? Can you add test that matches the GitLab bahvior?
- `^[section][2]` (From the doc "Optional sections ignore the number of approvals required")
- `[section] [2]`
- `[section] # [2]`

Reply:
Good catch, I'll have a look next week about those edge cases :)

Reply:
I've updated the parser and tests according to how GitLab handle those cases, as well as a few more similar ambiguities.